packages/contracts/test/scripts/MigrateMetapool.t.sol

```diff
@@ -0,0 +1,261 @@
+
+    IMetaPool v2Metapool =
+        IMetaPool(0x20955CB69Ae1515962177D164dfC9522feef567E);
+    /// Curve Metapool V3 Dollar:3CRV
+    IMetaPool v3Metapool =
+        IMetaPool(0x9558b18f021FC3cBa1c9B777603829A42244818b);
+
+    TWAPOracleDollar3pool twapOracle;
+
+    /// Forge Test Fork Identifier
+    uint256 mainnet;
+    /// Forge Snapshot selector
+    uint256 snapshot;
+
+    function setUp() public {
+        /*
+            Forking mainnet to run the tests and then taking a snapshot of the state so we can later revert to the snapshot for comparing states after two different logic runs.
+            Fork URL currently hardcoded due to current limits of Github Actions
+        */
+        mainnet = vm.createSelectFork("https://eth.ubq.fi/v1/mainnet");
+        snapshot = vm.snapshot();
+    }
+
+    function testMigrateCompare() public {
+        /// Ubiquity Dollar balance of V2 Metapool before migration
+        uint256 v2DollarPreBalance = dollarToken.balanceOf(address(v2Metapool));
+        /// Curve3 LP Balance of V2 Metapool before migration
+        uint256 v2Curve3PreBalance = curve3Token.balanceOf(address(v2Metapool));
+        /// Total amount of V2 Metapool LP tokens before migration
+        uint256 v2LP = v2Metapool.totalSupply();
+
+        (
+            uint256 metaBalance,
+            uint256 metaBalanceV3,
+            uint256 lpMinted
+        ) = _migrate();
+
+        /// Total amount of V3 Metapool LP tokens after migration
+        uint256 v3LP = v3Metapool.totalSupply();
+        /// Ubiquity Dollar balance of V3 Metapool after migration
+        uint256 v3DollarBalance = dollarToken.balanceOf(address(v3Metapool));
+        /// Curve3 LP balance of V3 Metapool after migration
+        uint256 v3Curve3Balance = curve3Token.balanceOf(address(v3Metapool));
+        /// Curve3 LP Balance of V2 Metapool after Migration
+        uint256 v2Curve3PostBalance = curve3Token.balanceOf(
+            address(v2Metapool)
+        );
+        /// Ubiquity Dollar balance of V2 Metapool after Migration
+        uint256 v2DollarPostBalance = dollarToken.balanceOf(
+            address(v2Metapool)
+        );
+
+        console.log("LP Tokens in Staking before migration: ", metaBalance);
+        console.log("LP Tokens in Staking after migration: ", metaBalanceV3);
+        console.log("Total LP Tokens Curve Metapool V2: ", v2LP);
+        console.log("Total LP Tokens Curve Metapool V3: ", v3LP);
+        console.log(
+            "Amount of Ubiquity Dollar Tokens in Curve Metapool V2: ",
+            v2DollarPreBalance
+        );
+        console.log(
+            "Amount of Ubiquity Dollar Tokens in Curve Metapool V2 after migration: ",
+            v2DollarPostBalance
+        );
+        console.log(
+            "Amount of Ubiquity Dollar Tokens in Curve Metapool V3: ",
+            v3DollarBalance
+        );
+        console.log(
+            "Amount of Curve3 Tokens in Curve Metapool V2 before migration: ",
```

Comment:
For example, I thought that these should be called something like:

- curve3PoolToken
- threePoolToken